src/main/java/org/wildfly/security/asn1/util/package-info.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * ASN.1 utilities.
```

Comment:
This isn't something I have asked anyone yet to do so far but where we have a package-info in a private package maybe we should consider adding a line like "The classes within this package are for the internal use of the WildFly Elytron project, changes to the APIs within this package may be made without preserving backwards compatibility."

That way if anyone depends on the main jar until we can get to Java 9 module definitions they at least have a chance of being notified ;-)  

Maybe for Java 8 we should consider publishing an API jar that can be developed against and an impl jar to be used for runtimes, that would at least give Java 8 developers a compile time check they are using public API only.

Anyway just some random thoughts I am throwing out there not really something to address in this PR ;-) I think we are in a good position to maintain compatibility in this project - just a few remaining steps to make it easier both for us and the third parties that depend on us.